Homework 18

hcarter3508 opened the pull request.

hcarter3508 committed:
Update Main.java

hcarter3508 committed:
Still has errors

Unable to locate Xpath to button once it is visible. The developer view is not showing it with the 'find' tool. Please advise.

Berichicko commented on src/test/java/Homework18.java:
```diff
@@ -0,0 +1,58 @@
+        // no response from the player otherwise. I even attempting the
+        // following line twice.... Please advise. The video for the following
+        // lecture is terrible.
+        clickXpathButton("//*[@id='mainFooter']/div[1]/i[2]");
```
**Use in test methods, the following logical split:   GIVEN – WHEN - THEN**
    
For example

```
 @Test

    public void testChangeProfileName (){
        // GIVEN

        String newName  = getRandomString();

        String currentPassword = "te$t$tudent";

        provideEmail("demo@class.com");

        providePassword();

        clickSubmitBtn();

 

        // WHEN

        openUserProfilePage();

        setName(newName);

        setCurrentPassword(currentPassword);

 

 

        // THEN

        Assert.assertTrue(getSuccessPopUp().isDisplayed());

}
```

Please, add Assert

**One test - One Assert**

Berichicko commented:
HINT

![18](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/12a4f3b4-8240-4c4b-8ffd-533394d464bb)

hcarter3508 commented:
data-testid works with By.cssSelector? Awesome! How did your get the xpaths from the developer view?

hcarter3508 commented:
You used an Action object. Nicely done. I was fixated on the WebElement.

Berichicko commented:
> data-testid works with By.cssSelector? Awesome! How did your get the xpaths from the developer view?

Hello

_For example_

`By.cssSelector("[data-testid='yourTestId']")`

`By.xpath("//*[@data-testid='yourTestId']")`


Please, re-submit homework 18 for review

hcarter3508 commented:
Is it better to throw an uncaught exception and the program die or to catch the exception, continue execution, and let the assertion fail?

Berichicko commented:
> Is it better to throw an uncaught exception and the program die or to catch the exception, continue execution, and let the assertion fail?

If an exception is caught and not properly handled or reported, it can lead to silent failures and unexpected behavior in the program.

hcarter3508 committed:
Finished

Noticed the use of the Action object. Also distinguished 'the next song' as the first in the list. Is this correct?
